### PR TITLE
[DUOS-2291][risk=no] Add create user info to select Dataset APIs

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2291

This PR adds the create user, when available, to the following dataset APIs:
* GET /api/dac/{dacId}/datasets
* GET /api/dataset/batch
* GET /api/dataset/search
* GET /api/dataset/v2
* GET /api/dataset/v2/{id}
* GET /api/tdr/{identifier}

Additionally, this PR cleans up a lot of duplicate and unused code.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java [hunk 1]
```diff
@@ -372,28 +403,12 @@ public void testGetDictionaryTerms() {
             assertNotNull(t.getKey());
         });
     }
-    @Test
-    public void testGetDataSetsForObjectIdList() {
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java:376]: Can you explain why some of these tests are no longer relevant? Is it because it was testing the duplicated codepaths / queries you mentioned? Also, what makes you confident that these should be deleted?

**Review by fboulnois — APPROVED** [review]
Overall, the inclusion of the user info makes sense to me. One comment around testing strategy as I familiarize myself with the codebase:

File: src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java (hunk 1 above)
Comment (rushtong) [reply src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java:376]: It's hard to tell from such a convoluted change list, but the underlying method under test here was removed so I removed the test method. This is something we unfortunately run into too often. We have extensive unit tests and sometimes the only usage of a method is in a test and not application code. I routinely prune these and look for this case with most PRs.

File: src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java [hunk 2]
```diff
@@ -872,4 +872,33 @@ private Dataset insertDataset() {
...
+    }
+
+    protected Consent insertConsent() {
+        String consentId = UUID.randomUUID().toString();
+        consentDAO.insertConsent(consentId,
+                false,
+            """
+                {"type":"everything"}
+                """,
+            """
```
Comment (connorlbark) [inline src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java]: nit: formatting

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 3]
```diff
@@ -558,12 +542,8 @@ public List<Dataset> findAllDatasetsByUser(User user) {
         }
     }
 
-    public Dataset getDataset(Integer datasetId) {
-        return this.datasetDAO.findDatasetById(datasetId);
-    }
-
     public List<Dataset> getDatasets(List<Integer> datasetIds) {
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: should this be renamed to `findDatasets`? it seems like the naming is out of convention, and it might make sense to update now since we're changing names in other places

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java [hunk 4]
```diff
@@ -57,33 +57,265 @@ Integer insertDataset(
         @Bind("dataUse") String dataUse,
         @Bind("dacId") Integer dacId);
 
+    @RegisterBeanMapper(value = User.class, prefix = "u")
     @UseRowReducer(DatasetReducer.class)
-    @SqlQuery(
-        "SELECT d.*, k.key, dp.property_value, dp.property_key, dp.property_type, dp.schema_property, dp.property_id, " +
-            " ca.consent_id, c.translated_use_restriction, dar_ds_ids.id as in_use, " +
-            FileStorageObject.QUERY_FIELDS_WITH_FSO_PREFIX +
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:64]: should we keep using the constant strings? now with use and fso both spelled out the query is very long / hard to parse, and maybe more brittle in case those change

Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:64]: (same thought goes for all the below DAO changes)

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java [hunk 5]
```diff
@@ -57,33 +57,265 @@ Integer insertDataset(
...
+        LEFT JOIN dataset_property dp ON dp.dataset_id = d.dataset_id
+        LEFT JOIN dictionary k ON k.key_id = dp.property_key
+        LEFT JOIN consent_associations ca ON ca.dataset_id = d.dataset_id
+        LEFT JOIN file_storage_object fso ON fso.entity_id = d.dataset_id::text AND fso.deleted = false
+        LEFT JOIN consents c ON c.consent_id = ca.consent_id
+        WHERE d.dataset_id = :datasetId
+    """)
     Dataset findDatasetById(@Bind("datasetId") Integer datasetId);
 
+    @RegisterBeanMapper(value = User.class, prefix = "u")
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java]: nit: maybe move this to the top of the class? so we don't need it on every call.

**Review by connorlbark — COMMENTED** [review]
looks great! couple nits, then 👍

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java (hunk 4 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:64]: So ... I had a similar concern. I tried a couple different ways of incorporating the query string constants into text blocks and they all either didn't work syntactically or looked horrible. I went this route because it was much easier to run the dev/test cycle with an open sql editor than trying to finagle query strings between each run.

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java (hunk 5 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java]: I didn't think that would work since it is not necessary for most of the other queries, but it turns out that it does, nice suggestion!

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java (hunk 3 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: I'll make that change - yes, it should. We have a fairly mixed bag of naming approaches here.

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java (hunk 4 above)
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:64]: that's fair - i think it would be worth figuring out how to do both, but we don't have to do that in this ticket. something to think about

**Review by connorlbark — APPROVED** [review]
👍
